base_layer/core/src/mining/miner.rs

```diff
@@ -86,21 +97,35 @@ impl Miner {
             utxo_sender,
             node_state_event_rx: None,
             mempool_state_event_rx: None,
+            miner_instruction_events,
             threads,
-            enabled: Arc::new(AtomicBool::new(false)),
+            mining_enabled_by_user: Arc::new(AtomicBool::new(false)),
+            mining_status: Arc::new(AtomicBool::new(false)),
             hashrate: Arc::new(AtomicU64::new(0)),
         }
     }
 
     /// This function instantiates a new channel and returns the receiver so that the miner can send out a unblinded
     /// output. This output is only sent if the miner successfully mines a block
-    pub fn get_utxo_receiver_channel(&mut self) -> Receiver<UnblindedOutput> {
+    pub fn get_utxo_receiver_channel(&mut self) -> mpscReceiver<UnblindedOutput> {
         // this should not be too large, as this should not get lost as these are your coinbase utxo's
-        let (sender, receiver): (Sender<UnblindedOutput>, Receiver<UnblindedOutput>) = mpsc::channel(20);
+        let (sender, receiver): (mpscSender<UnblindedOutput>, mpscReceiver<UnblindedOutput>) = mpsc::channel(20);
         self.utxo_sender = sender;
         receiver
     }
 
+    /// This function returns the sender portion of the mining instruction event channel so that start and
+    /// shutdown events can be sent to the miner while mining
+    pub fn get_miner_instruction_events_sender_channel(&self) -> syncSender<MinerInstruction> {
+        self.miner_instruction_events.clone()
+    }
+
+    /// This function returns the receiver portion of the mining instruction event channel so that start and
+    /// shutdown events can be received to interact with the miner while mining
+    pub fn subsrcibe_to_instruction_events(&self) -> syncReceiver<MinerInstruction> {
```

Comment:
@SWvheerden remarked that this can be called directly in `pub async fn mine(mut self)`. Agreed.